### PR TITLE
Add a new init template to setup a library

### PR DESCRIPTION
**Pull request by timotheeguerin**
Add a new library template with:
-  `@alernateName` decorator with some custom diagnostics  and tests for it.
- a linter rule and 2 rulesets(`recommended` and `all`)

### CODE REVIEW

File: packages/compiler/templates/__snapshots__/library-ts/lib/decorators.tsp [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+using TypeSpec.Reflection;
+
+namespace LibraryTs;
+
+/**
+ * Provide an alternate name for an operation.
+ * @param name The alternate name.
+ */
+extern dec alternateName(target: Operation, name: valueof string);
```
Comment (bterlson) [inline packages/compiler/templates/__snapshots__/library-ts/lib/decorators.tsp:11]: I wonder if there is risk that people think this decorator does something and can't delete it? Could rename to something obviously placeholder-y like "doSomething"? Just a thought.

Comment (bterlson) [reply packages/compiler/templates/__snapshots__/library-ts/lib/decorators.tsp:11]: Could also make the comment something like "An example decorator"

Comment (timotheeguerin) [reply packages/compiler/templates/__snapshots__/library-ts/lib/decorators.tsp:11]: Yeah I was trying to show a "real" use case but maybe I can prefix with `placeholder_`,  `example_` or `demo_`

File: packages/compiler/templates/__snapshots__/library-ts/src/decorators.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { DecoratorContext, Operation, Program } from "@typespec/compiler";
+import { createStateSymbol, reportDiagnostic } from "./lib.js";
+
+export const namespace = "LibraryTs";
+
+const alternateNameKey = createStateSymbol("alternateName");
```
Comment (bterlson) [inline packages/compiler/templates/__snapshots__/library-ts/src/decorators.ts:6]: Might be nice to put comments on the various parts here so it reads like a sample as well.

Comment (timotheeguerin) [reply packages/compiler/templates/__snapshots__/library-ts/src/decorators.ts:6]: I'll also change to the new way of doing state symbols

**Review by bterlson — APPROVED** [review]
Looks good, very badly want to see a monorepo template though.

**Comment by timotheeguerin** [comment]
> Looks good, very badly want to see a monorepo template though.

I can try adding that next

File: packages/compiler/templates/__snapshots__/library-ts/lib/decorators.tsp (hunk 1 above)
Comment (allenjzhang) [reply packages/compiler/templates/__snapshots__/library-ts/lib/decorators.tsp:12]: This is a great addition. There are some conventions we may have taken for granted such as `name` and `$name` in tsp and ts respectively. Also a brief intro about mechnesims various pieces in the projects can really help library writers with a short write-up in the doc section. We can provide a `@dev` section with the link in the comment section. Can we have a separate issue to take that if not in this PR?

**Review by allenjzhang — APPROVED** [review]
Looks good to me. Portal and other teams will greatly appreciate this. Thanks!

**Pull request merged**